nav2-usability/scenario-code/deeplink-pathparam/deeplink_pathparam_auto_route.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Deeplink path parameters example
+/// Done using AutoRoute
+import 'package:flutter/material.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter_uxr/main.gr.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+final List<Book> books = [
+  Book('Stranger in a Strange Land', 'Robert A. Heinlein'),
+  Book('Foundation', 'Isaac Asimov'),
+  Book('Fahrenheit 451', 'Ray Bradbury'),
+];
+
+class Book {
+  final String title;
+  final String author;
+
+  Book(this.title, this.author);
+}
+
+// Declare routing setup
+@MaterialAutoRouter(
+  replaceInRouteName: 'Screen,Route',
+  routes: <AutoRoute>[
+    AutoRoute(path: "/", page: BooksListScreen),
+    AutoRoute(path: "/book/:id", page: BookDetailsScreen),
+    RedirectRoute(path: "*", redirectTo: "/")
+  ],
+)
+class $AppRouter {}
+
+class BooksApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _appRouter = AppRouter();
+
+    return MaterialApp.router(
+      routerDelegate: _appRouter.delegate(),
+      routeInformationParser: _appRouter.defaultRouteParser(),
+    );
+  }
+}
+
+class BooksListScreen extends StatefulWidget {
+  @override
+  _BooksListScreenState createState() => _BooksListScreenState();
+}
+
+class _BooksListScreenState extends State<BooksListScreen> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: ListView(
+        children: [
+          for (var book in books)
+            ListTile(
+              title: Text(book.title),
+              subtitle: Text(book.author),
+              onTap: () =>
+                  context.router.pushPath("/book/${books.indexOf(book)}"),
```

Comment:
It's pushing a new one on top.l with a different key.

Reply:
Can we do a replace current page since we do that in the other libraries? I am also curious that whether auto-route do a push or replace in the RouterDelegate.setNewRoutePath method?

Reply:
Sorry - i missed that. I'll fix it after work.

Send part of your question I think Milad can answer

Reply:
@chunhtai actually it's doing a popUntil or push kinda thing, if the new page's key already exist in the stack s a pop untill will be performed + the anchor route will be updated if not a new entry is going to be pushed.
Eventually pressing the browser's back button feels like pressing the back button on android.

Reply:
Oh is it still using the navigator imperative API (push pop) to manage the stack? I think that is fine as long as it works, but just curious whether you have considered using the declarative API (Navigator.pages) to manage the stack?

Reply:
@chunhtai AutoRoute supports both declarative and imperative approaches, they can also be mixed, internally it's only using the pages API. 
for example you can do something like this

```dart
class _MyAppState extends State<MyApp> {
  final _appRouter = AppRouter();
  PageRouteInfo? _bookDetailsRoute;
  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      routerDelegate: AutoRouterDelegate.declarative(
        _appRouter,
        onInitialRoutes: (List<PageRouteInfo> routes) {
          // these routes are coming from setNewRoutePath
          var route = routes.first;
          if (route.routeName == BookDetailsRoute.name) {
            _bookDetailsRoute = route;
          }
        },
        routes: (context) => [
           const  BookListRoute(),
            if (_bookDetailsRoute != null) _bookDetailsRoute!,
          ],
      ),
}
```

Reply:
Changed to replace!